kong/conf_loader/init.lua

```diff
@@ -626,6 +656,63 @@ local function parse_value(value, typ)
 end
 
 
+-- Check if module is dynamic
+local function check_dynamic_module(mod_name)
+  local configure_line = ngx.config.nginx_configure()
+  local mod_re = [[^.*--add-dynamic-module=(.+\/]] .. mod_name .. [[(\s|$)).*$]]
+  return ngx.re.match(configure_line, mod_re, "oi") ~= nil
+end
+
+
+-- Lookup dynamic module object
+-- this function will lookup for the `mod_name` dynamic module in the following
+-- paths:
+--  - /usr/local/kong/modules -- default path for modules in container images
+--  - <nginx binary path>/../modules
+-- @param[type=string] mod_name The module name to lookup, without file extension
+local function lookup_dynamic_module_so(mod_name, kong_conf)
+  log.debug("looking up dynamic module %s", mod_name)
+
+  local mod_file = fmt("/usr/local/kong/modules/%s.so", mod_name)
+  if exists(mod_file) then
+    log.debug("Module '%s' found at '%s'", mod_name, mod_file)
+    return mod_file
+  end
+
+  local nginx_bin = nginx_signals.find_nginx_bin(kong_conf)
+  mod_file = fmt("%s/../modules/%s.so", pl_path.dirname(nginx_bin), mod_name)
+  if exists(mod_file) then
+    log.debug("Module '%s' found at '%s'", mod_name, mod_file)
+    return mod_file
+  end
+
+  return nil, fmt("%s dynamic module shared object not found", mod_name)
+end
+
+
+-- Validate Wasm properties
+local function validate_wasm(conf)
+  local wasm_enabled = conf.wasm
+  local filters_path = conf.wasm_filters_path
+
+  if wasm_enabled then
+    if filters_path and not exists(filters_path) and not isdir(filters_path) then
+      return nil, fmt("wasm_filters_path '%s' is not a valid directory", filters_path)
+    end
+  else
+    for cfg in pairs(conf) do
+      local wasm_cfg = match(cfg, "wasm_(.+)")
+      if wasm_cfg then
+        log.warn("Wasm is disabled but ", wasm_cfg,
```

Comment:
Yeah I think I lean slightly towards making this more quiet (change from warn to notice and/or only log once and then break out of the loop, etc) or omitting it entirely, but no strong opinion. I've adjusted the capitalization in another commit, so that's resolved.

@gszr authored this; I'll defer to him to decide if he wants to adjust (which I assume can be done post feature-freeze as well).

Reply:
Agreed. I will make that change in a follow-up PR.